### PR TITLE
Fix: Support mapping links to full URLs from `hint` Github repo

### LINKED ISSUE #933: Support mapping links to full URLs from `hint` GitHub repo
What the website is doing is getting these type of links:
```js
/* This regex should match the following examples:
     * * [link label]: ./somewhere.md → \s
     * * [link label]:./somewhere.md → :
     * * [link](./somewhere.md)  → \(
*/
```
And converting them to a "local" website.

If we add a full URL, for know it will going to keep it as is (IIRC). So the link instead of keeping inside the website, will go to the github documentation.

I think we need to add another rule to process urls that points to the hint repo.

_Originally posted by @sarvaje in https://github.com/webhintio/hint/pull/3937#issuecomment-673097734_

### PR DESCRIPTION
**Pull request by sarvaje**
Fix #933

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/webhint.io)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

### CODE REVIEW

File: helpers/update-site/documentation.js [hunk 1]
```diff
@@ -564,14 +569,25 @@ const updateMarkdownlinks = (content, filePath) => {
             val[0].replace(/(index|readme)\.md/i, '') :
             val[0].replace('.md', '/');
 
+        /*
+         * An github url is found for a md file.
+         * In this case we need to replace it with the a valid
+         * url in the website.
+         */
+        if (val[2].startsWith('https')) {
```
Comment (molant) [inline helpers/update-site/documentation.js:577]: It is a bit weird to say the link is for github and then check just for https.

```suggestion
        if (val[2].startsWith('https://github.com/')) {
```

File: helpers/update-site/documentation.js [hunk 2]
```diff
@@ -564,14 +569,25 @@ const updateMarkdownlinks = (content, filePath) => {
...
+            replacement =
+                `${val[1]}/docs/${file.frontMatter.section}${file.frontMatter.tocTitle ? `/${file.frontMatter.tocTitle.replace(' ', '-')}` : ''}/${val[4].replace('/docs/', '/')}/`;
+
```
Comment (molant) [inline helpers/update-site/documentation.js:580]: Maybe we could document what the different `val[x]` are?

**Comment by sarvaje** [comment]
@molant I have addressed all your feedback.

I also fix an edge case, when the package is the `hint` one.

**Comment by antross** [comment]
@molant any additional feedback or is this good to merge?

**Pull request merged**